File: pythonStuff/leetcode1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        outputList: int = []
        for x in range(len(nums)):
            
            #Check and see if the compliment of that number has been added to the dictonary
            compliment = target - nums[x]
            if str(compliment) in dict:
                outputList.append(dict[str(compliment)])
                outputList.append(x)
                return outputList
            #if the compliment for that number isnt found in the dictonary we add the number
            else:
                dict[str(nums[x])] = x
            

File: javaStuff/leetcode1.java
class Solution {
	public int[] twoSum(int[] nums, int target){
		Map<Integer, Integer> compliments = new HashMap<Integer, Integer>();
		int[] results = {0,0};
		for(int x = 0; x < nums.length; x++){
			int compliment = target - nums[x];
			if (compliments.containsKey(compliment)){
				results[0] = x;
				results[1] = compliments.get(compliment);
				return results;
			}
			else{
				compliments.put(nums[x],x);
			}
		}
		return results;		
	}
}
